src/sentry/integrations/slack/actions/notification.py

```diff
@@ -137,7 +216,51 @@ def send_notification(event: GroupEvent, futures: Sequence[RuleFuture]) -> None:
                     "rule.fail.slack_post",
                     extra=log_params,
                 )
-            rule = rules[0] if rules else None
+            else:
+                # Slack will always send back a ts identifier https://api.slack.com/methods/chat.postMessage#examples
+                # on a successful message
+                ts = None
+                # This is a workaround for typing, and the dynamic nature of the return value
+                if isinstance(response, BaseApiResponse):
+                    ts = response.json.get("ts")
+                elif isinstance(response, MappingApiResponse):
+                    ts = response.get("ts")
```

Comment:
if we get a `ts` back from these responses, can it be `None`?

Reply:
it can never be none, `ts` is a "guarantee" from slack haha
Unless slack decides to change their contract underneath our feet, but even so, it won't block future notifications since we make sure to filter out empty "ts" on the model (which is called "message_identifier")